src/main/java/edu/kit/ipd/crowdcontrol/objectservice/crowdworking/mturk/MturkRestCommand.java

```diff
@@ -0,0 +1,105 @@
+ */
+public abstract class MturkRestCommand<T,K> extends CompletableFuture<T> implements Callback<String> {
+    private final Function<K,T> transformer;
+    private final HttpRequest request;
+    private final Class<K> klass;
+    private final String uniqueRestToken;
+    private final String version;
+
+    /**
+     * Creates a new command
+     * @param con connection to use
+     * @param operation name of the operation to call
+     * @param responseGroup the group which should be returned by mturk
+     * @param version version of the api to use
+     * @param klass the class of the resulting structure
+     * @param supplier method to generate the parameters for the operation
+     * @param transformer the transformer which transforms the instance of klass into T
+     */
+    public MturkRestCommand(MTurkConnection con, String operation,
+                            String responseGroup, String version, Class<K> klass,
+                            Supplier<Map<String, Object>> supplier, Function<K, T> transformer) {
+        this.uniqueRestToken = UUID.randomUUID().toString().substring(0,20);
+        this.klass = klass;
+        this.transformer = transformer;
+        this.version = version;
+
+        request = Unirest.get(con.url)
+                .queryString("UniqueRequestToken", uniqueRestToken)
+                .queryString(con.getCallParameter(operation,responseGroup,version))
+                .queryString(supplier.get());
+        request.asStringAsync(this);
+    }
+    @Override
+    public void completed(HttpResponse<String> response) {
+        JAXBContext context;
+        try {
+            //FIXME we should append a version, but this leads to a bug ;)
+            final String bla = "http://requester.mturk.amazonaws.com/doc/2014-08-15";
+
+            NamespaceFilter filter =
+                    new NamespaceFilter(bla,true);
+            XMLReader reader
+                    = XMLReaderFactory.createXMLReader();
+            filter.setParent(reader);
+
+            context = JAXBContext.newInstance(klass);
+            SAXSource source = new SAXSource(filter, new InputSource(response.getRawBody()));
+
+            @SuppressWarnings("unchecked") K k = (K) context.createUnmarshaller().unmarshal(source);
+            complete(transformer.apply(k));
+        } catch (Exception e) {
+            completeExceptionally(new RuntimeException(e));
+        }
+    }
+
+    @Override
+    public void failed(UnirestException e) {
+        completeExceptionally(
+                getDetailedException(e)
+        );
+    }
+
+    /**
+     * Generates a detailed exception with unique id + the command
+     * @param e The exception which caused the problem
+     * @return returns a exception with unique id and command
+     */
+    private RuntimeException getDetailedException(UnirestException e) {
+        return new RuntimeException("Call "+uniqueRestToken+ " failed\n" +
```

Comment:
no its the intend of the exception since when this happens the call tree on the function should drop, print this exception.